pkg/ddl/affinity.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ddl
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/pkg/ddl/logutil"
+	"github.com/pingcap/tidb/pkg/domain/infosync"
+	"github.com/pingcap/tidb/pkg/meta/model"
+	"github.com/pingcap/tidb/pkg/parser/ast"
+	"github.com/pingcap/tidb/pkg/tablecodec"
+	pdhttp "github.com/tikv/pd/client/http"
+	"go.uber.org/zap"
+)
+
+// AffinityKeyRangeCodec encodes start/end keys to match the storage layout (e.g. keyspace aware).
+type AffinityKeyRangeCodec interface {
+	EncodeRegionRange(start, end []byte) ([]byte, []byte)
```

Comment:
Any reason to introduce a new interface instead of just use `codec.Codec` in the client-go?